datafusion/physical-plan/src/insert.rs

```diff
@@ -235,7 +243,9 @@ impl ExecutionPlan for DataSinkExec {
         let sink = Arc::clone(&self.sink);
 
         let stream = futures::stream::once(async move {
-            sink.write_all(data, &context).await.map(make_count_batch)
+            sink.write_all(data, &context)
+                .await
+                .map(|_| make_count_batch(num_rows as u64))
```

Comment:
Maybe not as part of this task but I'd rename the result column to match the snowflake result column name on insert and clarify it. From `count` to `number of rows inserted`.

Reply:
count is a field name, not sure we need column like 'number of rows inserted'. In any case if we want to be exactly like snowflake we must conduct a research. Works for me.